### ISSUE TITLE
BREAKING CHANGE: built-in tree-sitter support now requires a new-ish checkout of nixpkgs

### ISSUE THREAD

**Issue by leungbk** [nix-community/emacs-overlay#329]
See https://github.com/NixOS/nixpkgs/pull/230751 and https://github.com/nixos/nixpkgs/commit/1a8edfe192baf0775391ce520092db3c37ff5035 for how to install tree-sitter stuff using the new method. These commits are included in 23.05 and nixos-unstable, but are not in 22.11.

I removed support for the old way (which used overlay-only attributes) in https://github.com/nix-community/emacs-overlay/commit/509a7ef0ea8d63fac38a09f68fde3f4fe8678c92 and subsequently added a workaround that allowed our attributes to build on older checkouts of nixpkgs, albeit without tree-sitter stuff installed. If you want tree-sitter, then for newer revisions of this overlay, you will need to upgrade your nixpkgs accordingly.

I haven't checked whether or not we can have the old and new ways co-existing (maybe something weird happens to the load-path?), but if someone can confirm that it works, feel free to PR.

**Comment by jasonjckn** [comment]
TLDR ;

overlays = 
```
{ system, inputs, ... }:

[

  inputs.emacs-overlay.overlays.default

  # -------------------------------------------------------------------------------------------------------

  (final: prev: {

    # -------------------------------------------------------------------------------------------------------

    myEmacsGit =
      (prev.emacsPackagesFor
        ((prev.emacsGit.overrideAttrs
          (old: { withTreeSitter = true; }))

        )).emacsWithPackages (epkgs: with epkgs; [
          vterm
          (treesit-grammars.with-grammars
            (grammars: with grammars;
              [
                tree-sitter-bash
                tree-sitter-nix
              ]))
        ]) ;

    # -------------------------------------------------------------------------------------------------------
  })
]

```

**Comment by smaximov** [comment]
This worked for me:

```nix
    emacsWithPackagesFromUsePackage {
      # Your Emacs config file. Org mode babel files are also
      # supported.
      # NB: Config files cannot contain unicode characters, since
      #     they're being parsed in nix, which lacks unicode
      #     support.
      # config = ./emacs.org;
      config = emacs-config;
      package = emacs-pgtk;
      extraEmacsPackages = epkgs:
        [
          (epkgs.treesit-grammars.with-grammars (grammars:
            with grammars; [
              tree-sitter-bash
              tree-sitter-css
              tree-sitter-dockerfile
              tree-sitter-elixir
              tree-sitter-heex
              tree-sitter-html
              tree-sitter-javascript
              tree-sitter-json
              tree-sitter-nix
              tree-sitter-ruby
              tree-sitter-rust
              tree-sitter-toml
              tree-sitter-tsx
              tree-sitter-typescript
              tree-sitter-yaml
            ]))
        ];
    }
```
